src/routes/event-match.tsx

```diff
@@ -91,6 +94,36 @@ const EventMatch = ({ eventKey, matchKey }: Props) => {
     [match],
   )
 
+//  Variables used to assist in deciding if match key is parse-able 
+const parsedMatchType = m.group.replace(/\d+/g, '')
+const matchKeyFinalChar = m.group.charAt(m.group.length - 1)
+const numbers = ["1","2","3","4","5","6","7","8","9","0"]
```

Comment:
I think we had talked about this before, but can you remind me why it is beneficial to determine whether a match key is parseable?

How do we handle match keys that are parseable but are nonexistent matches? (i.e. `qm1000`)

Reply:
I'm more in favor of Solution 3. Coming from a user's perspective, they may find it difficult to interpret the error message if only a match key is provided for context. With that said, I'd **immensely** appreciate any suggestions that'll simplify the code (it has always been a bit convoluted to begin with 💫)

Reply:
Sounds good :+1: 

I think it makes the most sense to change the `formatMatchKey` function so that it no longer allows for invalid match keys. You can make it so that it returns `null` if the match key is invalid. I would change [this line](https://github.com/Pigmice2733/peregrine-frontend/blob/match-page-errors/src/utils/format-match-key/index.ts#L6) to return just `null` rather than the key in uppercase. Then in this file you can do:

```js
const parsedMatch = formatMatchKey(m.key)
```

and

```js
parsedMatch === null ? "This match does not exist" : /* ... */
```

Not completely sure this suggestion will work as-is, but it should at least be a good starting point. I have some free time later this week in case we want to do a call to work through this.